db/models/articles.model.js

```diff
@@ -36,10 +36,18 @@ exports.selectArticleComments = (article_id) => {
 
 exports.selectArticleById = (article_id) => {
   return db
-    .query(`SELECT * FROM articles WHERE article_id = $1`, [article_id])
-    .then((response) => {
-      const article = response.rows[0];
-      if (response.rows.length === 0)
+    .query(
+      `SELECT articles.author, articles.title, articles.article_id, articles.body, articles.topic, articles.created_at, articles.votes, articles.article_img_url, COUNT(comments.article_id) AS comment_count FROM articles
+    JOIN comments
+    ON articles.article_id = comments.article_id
+    WHERE articles.article_id = $1
+    GROUP BY articles.article_id, comments.article_id`,
+      [article_id]
+    )
+
+    .then(({ rows }) => {
+      const article = rows;
+      if (rows.length === 0)
         return Promise.reject({ status: 404, msg: "not found" });
 
       return article;
```

Comment:
As we are only returning one article here we can just return the object {} rather than the array [{}], which would then mean you don't have to access the first item in the array in your tests.

Reply:
altered code and test to make these changes.